src/sentry/projects/services/project/impl.py

```diff
@@ -178,3 +179,25 @@ def get_or_create_project_for_organization(
             add_org_default_team=add_org_default_team,
             external_id=external_id,
         )
+
+    def update_project(
+        self,
+        *,
+        organization_id: int,
+        project_id: int,
+        updates: ProjectUpdates,
+    ) -> RpcProject:
+        project: Project = Project.objects.get(
+            id=project_id,
+            organization_id=organization_id,
+        )
+        allowed_fields = set(ProjectUpdates.__annotations__.keys())
+        invalid_fields = [field for field in updates if field not in allowed_fields]
+        if invalid_fields:
+            raise ValueError(f"Invalid fields: {', '.join(invalid_fields)}")
+
+        for key, value in updates.items():
+            setattr(project, key, value)
+        project.save()
```

Comment:
There are some additional validation rules that we typically apply to project slug updates via the ProjectAdminSerializer https://github.com/getsentry/sentry/blob/2f5ae60d8252fb5b418da086442aea7a4600f17b/src/sentry/api/endpoints/project_details.py#L144-L148

should those be applied here as well?

Reply:
Could we simply re-use `ProjectAdminSerializer` here and set whatever fields the project has? Or make a separate serializer that does this job?

Reply:
I added a serializer for known fields, I hope this is what is expected